examples/SimpleDrone/SimpleDrone.py

```diff
@@ -0,0 +1,244 @@
+                                                    0, # X velocity in NED frame in m/s
+                                                    0, # Y velocity in NED frame in m/s
+                                                    0, # Z velocity in NED frame in m/s
+                                                    0, 0, 0, # afx, afy, afz acceleration (not supported yet, ignored in GCS_Mavlink)
+                                                    0, 0)    # yaw, yaw_rate (not supported yet, ignored in GCS_Mavlink)
+
+        # send command to vehicle
+        self.vehicle.send_mavlink(msg)
+        self.vehicle.flush()
+        self.last_send_translate = time.time()
+        self.point(0)
+        print "translating..."
+        if wait_for_arrival:
+            while self.vehicle.mode == "GUIDED":
+                veh_loc = self.vehicle.location.global_relative_frame
+                diff_lat_m = (lat - veh_loc.lat) * latlon_to_m
+                diff_lon_m = (lon - veh_loc.lon) * latlon_to_m
+                diff_alt_m = alt - veh_loc.alt
+                dist_xyz = math.sqrt(diff_lat_m**2 + diff_lon_m**2 + diff_alt_m**2)
+                if dist_xyz < dist_thres:
+                    print "Arrived"
+                    return
+            print '\033[93m' + "SimpleDroneError: 'translate({0},{1},{2})' was interrupted. Vehicle was switched out of GUIDED mode".format(x,y,z) + '\033[0m'
+
+    # takeoff - takeoff to some altitude, needs armed status - params: meters
+    def takeoff(self, altitude_meters=3):
+        if self.vehicle.mode != 'GUIDED':
+            print '\033[91m' + "SimpleDroneError: 'takeoff({0})' was not executed. Vehicle was not in GUIDED mode".format(altitude_meters) + '\033[0m'
+            return
+
+        while not self.vehicle.armed:
+            print " Waiting for arming..."
+            time.sleep(1)
+
+        print "Taking off..."
+        if self.vehicle.system_status != SystemStatus('STANDBY'):
+            print "Already airborne"
+            return
+        self.vehicle.simple_takeoff(altitude_meters)
+        # Wait until the vehicle reaches a safe height before processing the goto (otherwise the command
+        #  after Vehicle.simple_takeoff will execute immediately).
+        while self.vehicle.mode == 'GUIDED':
+            #Break and return from function just below target altitude.
+            if self.vehicle.location.global_relative_frame.alt>=altitude_meters*0.95:
+                print "Takeoff Complete"
+                return
+            time.sleep(1)
+        print '\033[93m' + "SimpleDroneError: 'takeoff({0})' was interrupted. Vehicle was swicthed out of GUIDED mode".format(altitude_meters) + '\033[0m'
+
+
+    # RTH - return to home
+    def RTH(self):
+        self.vehicle.mode = VehicleMode("RTL")
+
+    # enable_fence - Activate a safe operating zone - params: meters, meters, radius of cylinder in meters
+    def enable_fence(self, alt_floor, alt_ceiling ,radius):
```

Comment:
I agree that a geofence makes a great example of what you can do with dronekit, but perhaps better to create a new and innovative fence then have an implementation that simply duplicates what copter already does?

I think that we'd be better off in this case perhaps having your same API, but implementing it using the normal Copter fence parameters. This would then probably be part of a dronekit examples, not solo specific.

This does however point to the need for DK to have a fence API of some kind that would work across vehicles (Plane does fence quite differently and more flexibly).